clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientBuilderImpl.java

```diff
@@ -209,11 +215,45 @@ public ZeebeClientBuilder credentialsProvider(CredentialsProvider credentialsPro
     return this;
   }
 
+  @Override
+  public ZeebeClientBuilder oAuthCredentialsProvider(String clientId, String clientSecret) {
+    this.clientId = clientId;
+    this.clientSecret = clientSecret;
+    return this;
+  }
+
+  @Override
+  public ZeebeClientBuilder oAuthCredentialsProvider(
+      String clientId, String clientSecret, String authzServerUrl) {
+    this.authzServerUrl = authzServerUrl;
+    return oAuthCredentialsProvider(clientId, clientSecret);
+  }
+
   @Override
   public ZeebeClient build() {
+    applyDefaults();
+
     return new ZeebeClientImpl(this);
   }
 
+  private void applyDefaults() {
+    if (clientId != null && clientSecret != null) {
+      final Matcher matcher = PORT_PATTERN.matcher(brokerContactPoint);
+
+      if (matcher.find()) {
+        final String audience = brokerContactPoint.substring(0, matcher.start());
```

Comment:
Same as with the Go, I think we can go simpler with just `lastIndexOf` or whatever the name is than using Regex.